kubeflow/gcp/setup_backend.sh

```diff
@@ -22,9 +22,18 @@ gcloud auth activate-service-account --key-file=${GOOGLE_APPLICATION_CREDENTIALS
 gcloud config list
 
 NODE_PORT=$(kubectl --namespace=${NAMESPACE} get svc ${SERVICE} -o jsonpath='{.spec.ports[0].nodePort}')
+echo "node port is ${NODE_PORT}"
+
+while [[ -z ${BACKENDS} ]]; do
+  BACKENDS=$(kubectl --namespace=${NAMESPACE} get ingress ${SERVICE}-ingress -o jsonpath='{.metadata.annotations.ingress\.kubernetes\.io/backends}')
```

Comment:
using ${SERVICE}-ingress as ingress name is too fragile. Can we pass it in as a environment var?

Reply:
seems it will be different if using istio:
https://github.com/kubeflow/kubeflow/blob/48edfe412d0a3dd1dd4efa73a866dc5826c92336/kubeflow/gcp/iap.libsonnet#L386

could you show me the service name if using istio?

Reply:
The service after ingress is istio-ingressgateway if using Istio.
 I think we should just make ingress name as a param in iap.libsonnet, and pass it here as environment

Reply:
just to make sure: is it `envoy-ingress` for IAP or `istio-ingressgateway` for istio?

Reply:
actually, is it possible that you bring up a KF cluster with Istio so that I could update the logic with it?

Reply:
They are different things. You can check the iap.libsonnet:
1. the ingress name is always envoy-ingress (we can change, but doesn't matter)
2. the SERVICE (the service right behind ingress) is envoy for now, and istio-ingressgateway for iap.

Reply:
updated, please help me take a look, thanks!